coderd/database/dbauthz/dbauthz.go

```diff
@@ -3158,6 +3158,14 @@ func (q *querier) InsertUserLink(ctx context.Context, arg database.InsertUserLin
 
 func (q *querier) InsertWorkspace(ctx context.Context, arg database.InsertWorkspaceParams) (database.WorkspaceTable, error) {
 	obj := rbac.ResourceWorkspace.WithOwner(arg.OwnerID.String()).InOrg(arg.OrganizationID)
+	tpl, err := q.GetTemplateByID(ctx, arg.TemplateID)
+	if err != nil {
+		return database.WorkspaceTable{}, xerrors.Errorf("verify template by id: %w", err)
+	}
```

Comment:
This does check the template `read` perm again, but we cache all authz requests per `ctx`. So this check is going to be a cache lookup assuming it came from a normal code path.

Reply:
We only cache the authz decisions. The template fetch will happen again unfortunately (from a performance POV).